plugins/org.eclipse.xtend.core/src/org/eclipse/xtend/core/macro/AbstractFileSystemSupport.xtend

```diff
@@ -1,24 +1,34 @@
 package org.eclipse.xtend.core.macro
 
+import com.google.common.io.ByteStreams
 import com.google.common.io.CharStreams
 import com.google.inject.Inject
 import java.io.IOException
+import java.io.InputStream
 import java.io.InputStreamReader
 import java.io.UnsupportedEncodingException
+import java.net.URL
 import org.eclipse.emf.common.util.URI
 import org.eclipse.emf.ecore.resource.Resource
+import org.eclipse.emf.ecore.resource.ResourceSet
 import org.eclipse.xtend.lib.annotations.Accessors
 import org.eclipse.xtend.lib.macro.file.MutableFileSystemSupport
 import org.eclipse.xtend.lib.macro.file.Path
 import org.eclipse.xtext.generator.IFilePostProcessor
 import org.eclipse.xtext.parser.IEncodingProvider
 import org.eclipse.xtext.util.StringInputStream
-import org.eclipse.emf.ecore.resource.ResourceSet
+import org.eclipse.xtext.workspace.IWorkspaceConfigProvider
+
+import static org.eclipse.emf.ecore.resource.URIConverter.*
+import static org.eclipse.xtend.lib.macro.file.Path.*
+
+import static extension org.eclipse.xtext.util.UriUtil.*
 
 abstract class AbstractFileSystemSupport implements MutableFileSystemSupport {
```

Comment:
There's a lot of sneaky throw in the generated code for this class. The failure cases should all be handled / wrapped properly and documented on the API.